plumbing/hash/hash_sha1.go

```diff
@@ -1,15 +1,57 @@
-//go:build !sha256
-// +build !sha256
-
 package hash
 
-import "crypto"
+import (
+	"bytes"
+	"crypto"
+	"encoding/hex"
+	"io"
+)
 
 const (
 	// CryptoType defines what hash algorithm is being used.
 	CryptoType = crypto.SHA1
 	// Size defines the amount of bytes the hash yields.
-	Size = 20
+	Size = SHA1Size
 	// HexSize defines the strings size of the hash when represented in hexadecimal.
-	HexSize = 40
+	HexSize = SHA1HexSize
 )
+
+type SHA1Hash struct {
```

Comment:
Perhaps we could move the actual implementation in a sub package `plumbing/hash/sha1` with a unified API for both sha1 and sha256

I could also see the use for an object format registry similar to `crypto.RegisterHash`

Reply:
> I could also see the use for an object format registry similar to `crypto.RegisterHash`

I had some discussions with the folks from [gittuf](https://github.com/gittuf/gittuf) a while back, and that seemed to support some use cases they had. Let's definitely consider this.

Some of these changes I already have locally. I am thinking on `v6-transport` merged first, and then getting them going. Just to avoid all the rebasing.